addons/website_forum/models/forum_post.py

```diff
@@ -868,3 +868,34 @@ def _search_render_results(self, fetch_fields, mapping, icon, limit):
             if with_date:
                 data['date'] = self.env['ir.qweb.field.date'].record_to_html(post, 'write_date', {})
         return results_data
+
+    def _get_related_posts(self, limit=5):
+        """Return at most a list of 5 posts related to the main post, based on tag (Jaccard) similarity."""
+
+        if not self.tag_ids:
+            return self.env['forum.forum']
+
+        self.ensure_one()
+        self.env.cr.execute(SQL("""
+            SELECT forum_post.id,
+              -- Jaccard similarity
+                   (COUNT(DISTINCT commom_rel.forum_tag_id))::DECIMAL
+                   / COUNT(DISTINCT union_rel.forum_tag_id)::DECIMAL AS similarity
+              FROM forum_post
+              -- common tags (intersection)
+              JOIN forum_tag_rel AS commom_rel
```

Comment:
Typo, proposal of slight rename ?
```suggestion
              JOIN forum_tag_rel AS intersection_tag_rel
```